### DISCUSSION TITLE
parsing Time series data (latitude and longitude) from cassandra for worldmap panel

### DISCUSSION THREAD

**Discussion by iitm707** [HadesArchitect/GrafanaCassandraDatasource#146] [Ideas]
Hi,
I was trying to plot time series location data from my cassandra database on worldmap panel. But unfortunately current implementation does not supports my query because it gives error if i try to query latitude and longitude columns along with timestamp and id columns.
It will be great if custom queries can be made a little bit more flexible to allow such queries like i mentioned above..
Thank You 👍

**Comment by HadesArchitect** [comment]
Hi @iitm707,

I believe it's a pretty good idea and a feature nice to have. Could you please provide a sample data you have (like "describe table" statement + some rows) for me to make experiments with? :) Next thing, what's the full name / ID of the panel plugin you are working with?

@futuarmo FYI

**Reply by iitm707** [reply]
Hi,
Sorry for my late reply. I have successfully implemented what i desired as feature enhancement. 
I was working on [Worldmap Panel](https://grafana.com/grafana/plugins/grafana-worldmap-panel/) plugin and trying to use cassandra as a datasource.
Since this plugin required some extra columns for geo-coordinates, I modified your source code as per my requirement. and it works absolutely fine. :) cheers 👍  Meanwhile i also got to learn how everything works in the backend.

**Reply by iitm707** [reply]
I have added one more small enhancement which allows me to use variable in "ID Value" text field which can resolve this issue [#100 ](https://github.com/HadesArchitect/GrafanaCassandraDatasource/issues/100). Thanks

**Reply by futuarmo** [reply]
@iitm707 please make PR with your enhancement

**Reply by HadesArchitect** [reply]
Wow, nice! @iitm707 Indeed please submit a PR! It shouldn't be perfect, just let us see what you did and we could merge / adopt it by the project for the community to enjoy the new feature!

**Reply by iitm707** [reply]
> I have added one more small enhancement which allows me to use variable in "ID Value" text field which can resolve this issue [#100 ](https://github.com/HadesArchitect/GrafanaCassandraDatasource/issues/100). Thanks

I have created PR for this.

**Reply by iitm707** [reply]
I will create PR for feature enhancement to support [WorldMap Panel](https://grafana.com/grafana/plugins/grafana-worldmap-panel/) in next two days.

**Reply by brentellis** [reply]
@iitm707 did you ever publish the PR for this, I've got a requirement to present similar map data from Cassandra.

**Reply by iitm707** [reply]
@brentellis Adding this feature involved lots of changes in code.. I have implemented it as per my requirement and it is working fine.. main problem is that the number of columns and column types in this plugin are hard coded. I think this is due to limitations of Go API support for cassandra. So, if i merge my code with this, maybe the previous test cases of plugins will also need update, otherwise they will fail. Thats why I havent created PR for that.  I wished that a generic solution would have been there.. where anyone can add any type of columns and any number of columns to parse data from cassandra. I have already created one PR to resolve the issue #100 but it is still kept open for so long time. Dont know the reason for that..

**Reply by brentellis** [reply]
@iitm707 any chance you could still submit it as a PR so I can see it or send it to me?

**Reply by iitm707** [reply]
Ok I will submit the PR for this.. It might be helpful for others as well.. :)

**Reply by iitm707** [reply]
@HadesArchitect @futuarmo : Please review the code changes. I have created a PR for this

**Comment by HadesArchitect** [comment]
Hi @iitm707,

First of all, thanks a lot for your contribution! It's great to see your effort to improve the data source! We definitely should support the `worldmap` panel.

Although there is an issue in the offered implementation, something that doesn't look good to me. "latitude" and "longitude" are hard-coded everywhere, but they are particular panel specific, not general purpose. Every now and then, there will be a need to add support for some other panels, and we can't do it like this, just by adding more and more fields.

I'm adding @futuarmo and @unflag for their opinion.

**Reply by iitm707** [reply]
Hi @HadesArchitect ,
I also agree that this solution needs to be made more flexible to remove dependency on any hard coded parameters that i have used. 
we need to work on general purpose solution. I feel the Go API for cassandra are not that much flexible to allow this. That might be the main reason for it.

**Reply by unflag** [reply]
Hi everyone! Can't say if this is possible as I haven't looked into the code yet, but I too agree that such feature should be added as a general purpose solution. I'll find some time to understand the problem and return to discuss possible implementations.

**Reply by unflag** [reply]
Finally got a bit of time to do some investigation, sorry for waiting. Now I have a few thoughts on that matter.
* Plugin at its current state could not be used as Table Data source, because it supports only TimeSeries format. However, Table support would be a nice feature and I do not see any blockers for adding it except a lack of free time:) So, in other words, to support WorldMap panel only Table support has to be implemented and it is a general purpose solution.
* WorldMap is deprecated and there is a recommendation to use a [Geomap panel](https://grafana.com/docs/grafana/latest/panels-visualizations/visualizations/geomap/) instead. I am not sure if these two plugins are similar in terms of datasources and data formats they are using, so it should be checked out too.

**Comment by HadesArchitect** [comment]
@unflag This one is done and gone, right?

**Reply by unflag** [reply]
That's right. Table support was added in #167
